src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -821,10 +821,10 @@ void PersistNode(TrieNode n)
             });
         }
 
-        public byte[]? this[byte[] key]
+        public byte[]? this[ReadOnlySpan<byte> key]
         {
             get => _pruningStrategy.PruningEnabled
-                   && _dirtyNodes.AllNodes.TryGetValue(new Keccak(key), out TrieNode? trieNode)
+                   && _dirtyNodes.AllNodes.TryGetValue(new Keccak(key.ToArray()), out TrieNode? trieNode)
```

Comment:
This is my general take on this PR in regards to using `ConcurrentDictionary<KeccakKey>` instead of the Span based dictionary which was aimed (at least from the issues that you linked @LukaszRozmej ). Let's bench.